src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormArtworkStep.tsx

```diff
@@ -57,12 +57,29 @@ export const MyCollectionArtworkFormArtworkStep: React.FC<MyCollectionArtworkFor
 
     // Initialize photos with artwork images
 
-    const photos = artwork.images?.map(image => ({
-      name: "Automatically added",
-      url: image?.imageURL?.replace(":version", "large"),
-      id: image?.internalID,
-      size: image?.width,
-    }))
+    const photos = artwork.images?.map(image => {
+      const imageVersionsSortedBySize = [
+        "main",
+        "normalized",
+        "larger",
+        "large",
+        "medium",
+        "small",
+      ]
+
+      const imageVersion = imageVersionsSortedBySize.find(version =>
+        image?.imageVersions?.includes(version)
+      )
+
+      return {
+        name: "Automatically added",
+        url: !!imageVersion
+          ? image?.imageURL?.replace(":version", imageVersion)
+          : null,
```

Comment:
Could we pass the list of versions directly in the GrqphQL quer instead ([example](https://github.com/artsy/force/blob/b64d9edf38df65a3a2733799771cd2f09c131e4d/src/Apps/Article/Components/Sections/ArticleSectionImageCollection/ArticleSectionImageCollectionImage.tsx#L66))? I think this case should be supported 🤔

Reply:
didn't know we can do that! does this pattern also return the first one available in the provided array?

Reply:
Yes, I think it does exactly what we need. It goes through `imageVersions` and looks for the first version that exists ([code](https://github.com/artsy/metaphysics/blob/eb9a1362323222a5bd22089c4b798f1442af4a89/src/schema/v2/image/normalize.ts#L24)).

You should be able to get the image URL like this:

```graphql
image {
  url(
    version: [
        "main",
        "normalized",
        "larger",
        "large",
        "medium",
        "small"
    ]
  )
}
```

Reply:
great, thanks @olerichter00. I will update

Reply:
it seems like the version string is not supported within the Image interface in MP. I don't believe it makes sense to add support to it in this context, what do you think?

Reply:
I updated the query in `MyCollectionArworkSearch.tsx` and it appears to be working correctly.
```
images {
                  height
                  internalID
                  isDefault
                  url(version: ["large", "tall"])
                  width
                }
```

Reply:
I used `imageUrl` initially , i can try this out as well 👀

Reply:
worked! thanks